plugins/modules/terraform.py

```diff
@@ -324,6 +321,21 @@ def is_attribute_sensitive_in_providers_schema(
             if resource_schema_name == resource.type:
                 sensitive = resource_schema.attributes[attribute].sensitive
                 return sensitive
+
+    return False
+
+
+def is_blocktype_sensitive_in_providers_schema(
+    schemas: TerraformProviderSchemaCollection, resource: TerraformRootModuleResource, blocktype: str, subattribute: str
+) -> bool:
+    for provider_schema in schemas.provider_schemas:
+        resource_schemas = schemas.provider_schemas[provider_schema].resource_schemas
+        for resource_schema_name, resource_schema in resource_schemas.items():
+            if resource_schema_name == resource.type:
```

Comment:
I'm looking at this and realizing there's a bug in the existing code that is replicated here. The logic in this function and in `is_attribute_sensitive_in_providers_schema()` don't consider which provider a resource comes from. For example, `provider_a.attribute_foo` would also match `provider_b.attribute_foo` if both providers are used in the same config. Whichever provider schema showed up first would win. This should generally be pretty rare, since most providers namespace their resource names already, but it's possible.

I'm not sure there's a need to loop through provider schemas anyways, as the resource already has the provider schema name. The same goes for resource schemas. You should be able to just do:
```python
resource_schemas = schemas.provider_schemas[resource.provider_name].resource_schemas
resource_schema = resource_schemas[resource.type]
```